src/main/scala/beam/agentsim/scheduler/BeamAgentScheduler.scala

```diff
@@ -181,10 +169,12 @@ class BeamAgentScheduler(val beamConfig: BeamConfig, stopTick: Double, val maxWi
       }
       val completionTickOpt = triggerIdToTick.get(triggerId)
       if (completionTickOpt.isEmpty || !triggerIdToTick.contains(triggerId) || !awaitingResponse
-            .containsKey(completionTickOpt.get)) {
+        .containsKey(completionTickOpt.get)) {
         log.error(s"Received bad completion notice $notice from ${sender().path}")
       } else {
-        awaitingResponse.remove(completionTickOpt.get, triggerIdToScheduledTrigger(triggerId))
+        val st = triggerIdToScheduledTrigger(triggerId)
+        awaitingResponse.remove(completionTickOpt.get, st)
```

Comment:
Using the stuckFinder makes awaitingResponse unnecessary unless I missed something. Shouldn't we JUST use that instead?

Reply:
Good point. It seems to me that `StuckFunder` can cover the cases where `awaitingResponse` is used. @colinsheppard ?